pkg/controller/render/render_controller.go

```diff
@@ -487,9 +487,13 @@ func (ctrl *Controller) syncGeneratedMachineConfig(pool *mcfgv1.MachineConfigPoo
 		return err
 	}
 
-	// Emit an event so it's more visible that OSImageURL was overridden.
+	// Emit event and collect metric when OSImageURL was overridden.
 	if generated.Spec.OSImageURL != cc.Spec.OSImageURL {
```

Comment:
- I think this is totally fine where you put this, but I will mention that as part of #3317 I was annotating the machineconfig when we've overridden `OSImageURL` so other parts of the MCO can cheaply figure it out (because once we're out of the part of the render controller that does the override, everything else that wanted to know would have to compare vs controllerconfig to "guess" if we overrode, and that could result in complexity/races/synchronization, etc)
https://github.com/openshift/machine-config-operator/blob/e1bb09ad5c175180a1ce7fb81650f7af4eb29cb7/pkg/controller/render/render_controller.go#L588-L590
- So if you were to use that annotation to detect the condition, and you wanted to "corral the metrics together", you could probably stuff this new metric together near the other one in node_controller's pool sync, e.g. somewhere around here: https://github.com/openshift/machine-config-operator/blob/e1bb09ad5c175180a1ce7fb81650f7af4eb29cb7/pkg/controller/node/node_controller.go#L790 and read the annotation off the config when the pool syncs -- then you'd get the metric when it was assigned as pool spec and not just when it was rendered?

Reply:
no strong opinion but with your reasoning will tend towards implementing what you are suggesting. Will update the code accordingly. This would also mean that we will have to wait for MCO-289 to get merged first.

Reply:
on the second thought, I would say let's merge it as it is. We can do follow-up PR once PR#3317 gets merged and reading annotation makes it better. This will keep us unblocked for making further changes needed to be done in cluster monitoring operator and rhobs.